src/XIVLauncher.Common/Support/DalamudAndPluginConfigImportExport.cs

```diff
@@ -0,0 +1,121 @@
+
+            AddIfExist(accountsListFile, archive);
+            AddIfExist(dalamudConfigFile, archive);
+            AddIfExist(dalamudVfsFile, archive);
+            AddIfExist(pluginConfigsFolder, archive);
+
+            // add some known special exceptions. It might be better to not build these expectations though
+            var backupsFolder = Path.Combine(storagePath, "backups"); // Otter plugins
+            var playerTrackBackupsFolder = Path.Combine(storagePath, "playerTrackBackups"); // PlayerTrack
+
+            AddIfExist(backupsFolder, archive);
+            AddIfExist(playerTrackBackupsFolder, archive);
+
+            // return the folder containing our exported settings
+            return outFile.FullName;
+        }
+
+        public static void ImportConfig(string zipFilePath, string storagePath)
+        {
+            // grab any platform-specific details
+            // there's none to grab currently :)
+
+            // TODO: Decide if we're going to alert on overwriting config.
+            // Right now, Franz decided against it. The user has to intentionally try to use this feature
+            // Also, .Net Framework is dumb and will explode if we use ZipArchive.ExtractToDirectory()
+            // and there are any file conflicts. .Net Core doesn't have this issue though and provides
+            // an override we could have used to do it anyways. Alternatively, we could just delete
+            // all of the files/folders we'd be restoring to first, but that also feels bad.
+
+
+
+            var inFile = new FileInfo(zipFilePath);
+            using var archive = ZipFile.Open(inFile.FullName, ZipArchiveMode.Read);
+
+            // If we weren't on .Net Framework, we could use this...
+            // ZipFileExtensions.ExtractToDirectory(archive, storagePath, true);
+
+            foreach (var entry in archive.Entries)
+            {
+                var extractPath = storagePath + "\\" + entry.FullName;
+                // If we were going to warn about overwriting files, it would go here.
+                /*
+                bool promptAlwaysForOverwrite = true;
+                if (promptAlwaysForOverwrite && File.Exists(extractPath))
+                {
+                    // Make some prompt. Overwrite? Yes, Yes to All, Cancel
+
+                    if (result == no) break or something.
+                    if (result == yestoall) promptAlwaysForOverwrite = false;
+                    // yes is the default and needs no special handling.
+                }
+                */
+                if (!Directory.Exists(Path.GetDirectoryName(extractPath)))
+                {
+                    Directory.CreateDirectory(Path.GetDirectoryName(extractPath));
+                }
+                ZipFileExtensions.ExtractToFile(entry, extractPath, true);
+            }
+        }
+
+        private static void AddIfExist(string entryPath, ZipArchive zip)
```

Comment:
We already have this function in the tspack code. Maybe move to common util class?
https://github.com/goatcorp/FFXIVQuickLauncher/blob/6f388fb7787952e837b851b70dbec9ab8349eb3e/src/XIVLauncher/Support/PackGenerator.cs#L39

Reply:
This one's a little different from tspack code. I had to get a little hacky for directory handling. But that could probably still be used for tspacks too. I've moved it to a util method in xivlauncher.common.util, but the WPF app didn't really like me doing that and VS stopped giving me intellisense when I tried. I might need some handholding to get that moved properly. (It could just be my machine being annoying)